android/src/main/java/com/reactnativesecurekeystore/CipherBoxImpl.kt

```diff
@@ -0,0 +1,54 @@
+package com.reactnativesecurekeystore
+
+import android.security.keystore.KeyProperties
+import com.reactnativesecurekeystore.dto.EncryptedOutput
+import java.security.Key
+import java.security.MessageDigest
+import java.security.PrivateKey
+import java.security.Signature
+import javax.crypto.Cipher
+import javax.crypto.spec.GCMParameterSpec
+
+
+const val CIPHER_ALGORITHM =
+  "${KeyProperties.KEY_ALGORITHM_AES}/${KeyProperties.BLOCK_MODE_GCM}/${KeyProperties.ENCRYPTION_PADDING_NONE}"
+const val GCM_TAG_LEN = 128
+const val SIGN_ALGORITHM = "SHA256with${KeyProperties.KEY_ALGORITHM_RSA}"
+const val HMAC_ALGORITHM = "SHA256"
+
+class CipherBoxImpl : CipherBox {
+  override fun encryptData(key: Key, data: String): EncryptedOutput {
+    val cipher = Cipher.getInstance(CIPHER_ALGORITHM)
+    cipher.init(Cipher.ENCRYPT_MODE, key)
+
+    val encryptedData = cipher.doFinal(data.toByteArray());
+
+    return EncryptedOutput(encryptedData, cipher.iv)
+  }
+
+  override fun createSignature(key: PrivateKey, data: String): Signature {
+    val hash = data.toByteArray(charset("UTF8"))
+    val signature = Signature.getInstance(SIGN_ALGORITHM).run {
+      initSign(key)
+      update(hash)
+      this
+    }
+
+    return signature
+  }
+
+  override fun generateHmacSha(data: String): ByteArray {
+    val messageDigest = MessageDigest.getInstance(HMAC_ALGORITHM)
```

Comment:
I hope we know that HMAC SHA256 is different from SHA256. This implementation is simply calculating SHA256 but not HMAC SHA256. HMAC would require a AES key additionally.